src/utils/tests/test_cfgreader.py

```diff
@@ -0,0 +1,153 @@
+import pytest
+
+from utils.cfgreader import *
+
+
+unset_env_values = [None, '', ' ', '   ', '\t', '\n', '\r\n', '\n\r', ' \t ']
+
+MY_VAR = 'MY_VAR'
+
+unset_env_var = [{MY_VAR: v} for v in unset_env_values] + [{}]
+
+
+@pytest.mark.parametrize('value, expected', [
+    (None, ''), ('', ''),
+    (' ', ''), ('\t', ''), ('\n', ''), ('\r\n', ''),
+    (' x', 'x'), ('\tx ', 'x'), ('x\n', 'x'), (' x ', 'x')
+])
+def test_str_var(value, expected):
+    var = StrVar('V', '')
+    assert var.read(value) == expected
+
+
+@pytest.mark.parametrize('value', unset_env_values)
+def test_str_var_default(value):
+    def_val = 'some default value'
+    var = StrVar('V', def_val)
+    assert var.read(value) == def_val
+
+
+@pytest.mark.parametrize('value, expected', [
+    (' 5', 5), ('\t5 ', 5), ('5\n', 5), (' 5 ', 5),
+    (' 5432', 5432), ('\t5432 ', 5432), ('5432\n', 5432), (' 5432 ', 5432)
+])
+def test_int_var(value, expected):
+    var = IntVar('V', None)
+    assert var.read(value) == expected
+
+
+@pytest.mark.parametrize('value', unset_env_values)
+def test_int_var_default(value):
+    def_val = 12345
+    var = IntVar('V', def_val)
+    assert var.read(value) == def_val
+
+
+@pytest.mark.parametrize('value, expected', [
+    (' 5', False), ('\t5 ', False), ('5\n', False), (' 5 ', False),
+    ('false', False),
+    ('t', True), (' t', True), ('\tt ', True), ('t\n', True), (' t ', True),
+    ('T', True), (' T', True), ('\tT ', True), ('T\n', True), (' T ', True),
+    ('1', True), (' 1', True), ('\t1 ', True), ('1\n', True), (' 1 ', True),
+    ('yes', True), (' yes', True), ('\tyes ', True), ('yes\n', True),
```

Comment:
out of curiosity, who defines these valid options for `true`? E.g., why is `y` not there or why there shouldn't be?

Reply:
I've asked the interwebs for the most common ways to map a truthy string to Python's `True` and the answer was: 'true', 'yes', '1', 't'. This is kinda arbitrary anyway, so should I add `y` too?

Reply:
added `y` too :-)